apps/challenges/views.py

```diff
@@ -340,3 +357,161 @@ def challenge_phase_split_list(request, challenge_pk):
     serializer = ChallengePhaseSplitSerializer(result_page, many=True)
     response_data = serializer.data
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['POST'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def create_challenge_using_zip_file(request, challenge_host_team_pk):
+    """
+    Creates a challenge using a zip file.
+    """
+    try:
+        challenge_host_team =  ChallengeHostTeam.objects.get(pk=challenge_host_team_pk)
+    except:
+        response_data = {'error': 'Challenge Host Team for {} does not exist'.format(request.user)}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    serializer = ChallengeConfigurationSerializer(data=request.data, context={'request': request})
+    if serializer.is_valid():
+        uploaded_zip_file = serializer.save()
+        uploaded_zip_file_path = serializer.data['zip_configuration']
+    else:
+        response_data = serializer.errors
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        with transaction.atomic():
```

Comment:
We use `transaction.atomic` whenever we want to execute a set of more than db queries as a whole. Explaining this further, it means that we want either the whole set of query should be executed, or None if any of them fails.

Here we have such a use case, but i am really confused of using transaction.atomic to handle code related to http requests(like get request for zip file).

Can you please explain me your reason of using `transaction.atomic` in such a way?